### PR TITLE
travis-ci support

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

**Comment by agentzh** [comment]
@chipitsine Thank you for your hard work! I've merged a modified version of your patch to master. I made the following changes:

1. make luajit and nginx build in parallel (controlled by the `JOBS` env, I set it to 3 which seems to be the best).
2. enable assertions and API checks in the luajit build.
3. use the `v2.1-agentzh` branch instead of `v2.1` of the luajit2 repository since the former is the one used by OpenResty.
4. ensure that luajit build honor the `CC` env (yes, LuaJIT's Makefile also defaults to `gcc` instead of `$CC`).
5. install our own luajit into `/opt/luajit21/` instead of polluting the system library paths. this also makes verification easier and cleaner.
6. install the Perl CPAN module dependencies of `test-nginx` via `apt-get` such that `cpanm` does not have to install them itself (it's usually much faster to install the debian packages).
7. prepending new directories to the `PATH` env instead of appending, so that we can ensure we won't pick up anything unwanted in the existing `PATH` env value.
8. enable the debug symbols in our luajit build via the make argument `CCDEBUG=-g`.
9. split `make install` of luajit into 2 steps: `make` and `make install`.

I think we can just use this `.travis.yml` as the template for other OpenResty repositories :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  global:
+    - LUAJIT_LIB=/usr/lib64/libluajit-5.1.so
+    - LUAJIT_INC=/usr/include/luajit-2.1
+    - LUA_INCLUDE_DIR=/usr/include/luajit-2.1
+    - LUA_CMODULE_DIR=/lib
+  matrix:
+    - NGINX_VERSION=1.9.15
+    
+before_install:
+  - sudo apt-get install -qq -y axel cpanminus > build.log 2>&1 || (cat build.log && exit 1)
```
Comment (chipitsine) [inline .travis.yml]: "apt-get install -qq" is pretty silent itself, I wouldn't add output redirection here

**Comment by chipitsine** [comment]
I would also move luajit2 to ..
it produces to much noise with cppcheck (and we do not want to check luajit while compiling some nginx module)

**Comment by chipitsine** [comment]
I would also move luajit2 to ..
it produces to much noise with cppcheck (and we do not want to check luajit while compiling some nginx module)

**Comment by chipitsine** [comment]
I installed luajit into system dir, because I was lazy and I wanted to finish it quickly. If we specify custom dir, so we can get rid of sudo and install it somewhere in home.

Travis also supports docker images with "sudo: false" for such purpose

**Comment by agentzh** [comment]
@chipitsine cppcheck may have false positives. Maybe make it nonfatal when it finds anything.

Yes, better raise the component directories one level up. Patches welcome.

I'm not against installing under `/home/`, `/opt/` is just a common practice of myself.

**Pull request closed without merging**